openeo/extra/job_management.py

```diff
@@ -404,6 +412,34 @@ def on_job_error(self, job: BatchJob, row):
             self.ensure_job_dir_exists(job.job_id)
             error_log_path.write_text(json.dumps(error_logs, indent=2))
 
+    def on_job_cancel(self, job: BatchJob, row):
+        """
+        Handles jobs that that were cancelled. Can be overridden to provide custom behaviour.
+
+        Default implementation does not do anything.
+
+        :param job: The job that has finished.
+        :param row: DataFrame row containing the job's metadata.
+        """
+        pass
+
+    def _cancel_prolonged_job(self, job: BatchJob, row):
+        """Cancel the job if it has been running for too long."""
+        job_running_start_time = rfc3339.parse_datetime(row["running_start_time"], with_timezone=True)
+        current_time = rfc3339.parse_datetime(rfc3339.utcnow(), with_timezone=True)
+
+        
+        if current_time > job_running_start_time + self.cancel_running_job_after:
+            try:
+                print(str(self.cancel_running_job_after))
+                _log.info(
+                f"Cancelling job {job.job_id} as it has been running for more than {str(self.cancel_running_job_after)}"
```

Comment:
```suggestion
                f"Cancelling job {job.job_id} as it has been running for more than {self.cancel_running_job_after}"
```

the `{}` construct normally does the right thing automatically without explicit "str", but I might be mistaken about timedelta. Feel free to ignore